File: rust/src/191.rs
//191. Number of 1 Bits
impl Solution {
    pub fn hamming_weight(n: i32) -> i32 {
        let mut count = 0;
        let binary = format!("{:b}", n); // Convert to binary string

        for bit in binary.chars() {
            if bit == '1' {
                count += 1;
            }
        }

        count
    }
}

//sub: https://leetcode.com/problems/number-of-1-bits/submissions/1566738416/
//sol: https://leetcode.com/problems/number-of-1-bits/solutions/6512204/191-number-of-1-bits-by-nalindalal2004-5unl

File: ts/src/191.ts
//162. Find Peak Element
function findPeakElement(nums: number[]): number {
  if (nums.length === 0) return -1;

  let maxIndex: number = 0;
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] > nums[maxIndex]) maxIndex = i;
  }

  return maxIndex;
}
//sub: https://leetcode.com/problems/find-peak-element/submissions/1589635247/?envType=study-plan-v2&envId=top-interview-150
//sol: https://leetcode.com/problems/find-peak-element/solutions/6591239/162-find-peak-element-by-nalindalal2004-mcpc/
